include/rmm/mr/device/system_memory_resource.hpp

```diff
@@ -0,0 +1,136 @@
+ *     NVIDIA Ada Lovelace, NVIDIA Hopper, or newer.
+ *   - A 64-bit x86 CPU.
+ *
+ *  For more information, see
+ *  https://developer.nvidia.com/blog/simplifying-gpu-application-development-with-heterogeneous-memory-management/.
+ *
+ *  Address Translation Services (ATS) is a hardware/software solution for the Grace Hopper
+ *  Superchip that uses the NVLink Chip-2-Chip (C2C) interconnect to provide coherent memory. For
+ *  more information, see
+ *  https://developer.nvidia.com/blog/nvidia-grace-hopper-superchip-architecture-in-depth/.
+ */
+class system_memory_resource final : public device_memory_resource {
+ public:
+  system_memory_resource()
+  {
+    RMM_EXPECTS(rmm::mr::detail::sam::is_supported(rmm::get_current_cuda_device()),
+                "System memory allocator is not supported with this hardware/software version.");
+  }
+  ~system_memory_resource() override                    = default;
+  system_memory_resource(system_memory_resource const&) = default;  ///< @default_copy_constructor
+  system_memory_resource(system_memory_resource&&)      = default;  ///< @default_copy_constructor
+  system_memory_resource& operator=(system_memory_resource const&) =
+    default;  ///< @default_copy_assignment{system_memory_resource}
+  system_memory_resource& operator=(system_memory_resource&&) =
+    default;  ///< @default_move_assignment{system_memory_resource}
+
+ private:
+  /**
+   * @brief Allocates memory of size at least \p bytes.
+   *
+   * The returned pointer will have at minimum 256 byte alignment.
+   *
+   * The stream argument is ignored.
+   *
+   * @param bytes The size of the allocation
+   * @param stream This argument is ignored
+   * @return void* Pointer to the newly allocated memory
+   */
+  void* do_allocate(std::size_t bytes, [[maybe_unused]] cuda_stream_view stream) override
+  {
+    auto const aligned{rmm::align_up(bytes, rmm::CUDA_ALLOCATION_ALIGNMENT)};
+    return malloc(aligned);
+  }
+
+  /**
+   * @brief Deallocate memory pointed to by \p p.
+   *
+   * The stream argument is ignored.
+   *
+   * @param ptr Pointer to be deallocated
+   * @param bytes This argument is ignored
+   * @param stream This argument is ignored
+   */
+  void do_deallocate(void* ptr,
+                     [[maybe_unused]] std::size_t bytes,
+                     [[maybe_unused]] cuda_stream_view stream) override
+  {
+    free(ptr);
+  }
+
+  /**
+   * @brief Compare this resource to another.
+   *
+   * Two system_memory_resources always compare equal, because they can each deallocate memory
+   * allocated by the other.
+   *
+   * @param other The other resource to compare to
+   * @return true If the two resources are equivalent
+   * @return false If the two resources are not equal
+   */
+  [[nodiscard]] bool do_is_equal(device_memory_resource const& other) const noexcept override
+  {
+    return dynamic_cast<system_memory_resource const*>(&other) != nullptr;
+  }
+};
```

Comment:
For the new interfaces, you should define the accessibility properties: 

```suggestion
  /**
   * @brief Enables the `cuda::mr::device_accessible` property
   *
   * This property declares that a `system_memory_resource` provides device-accessible memory
   */
  friend void get_property(system_memory_resource const&, cuda::mr::device_accessible) noexcept {}

  /**
   * @brief Enables the `cuda::mr::device_accessible` property
   *
   * This property declares that a `system_memory_resource` provides host-accessible memory
   */
  friend void get_property(system_memory_resource const&, cuda::mr::host_accessible) noexcept {}
};

// static property checks
static_assert(cuda::mr::async_resource_with<system_memory_resource, cuda::mr::device_accessible>);
static_assert(cuda::mr::async_resource_with<system_memory_resource, cuda::mr::host_accessible>);
```